### ISSUE TITLE
VS Code does not send a focused event on native context menu close

### ISSUE THREAD

**Issue by jvesouza** [microsoft/vscode#93226]
Issue Type: <b>Bug</b>

1. Launch orca.
2. Launch VSCode and create a file called index.js with the following content:
>var x = 0;
>x = 10;
3. Move the cursor to the second line by placing it over the variable x used in the assignment.
4. Press f12 (go to definition).

The cursor will be moved to the variable declaration and we can use the arrows without problems to navigate the text as expected.

5. Repeat the step 3.
6. Press the application key and choose go to definition.
7. Try to navigate using up/down arrow.

At least in my environment orca announces the name of the file being edited every time I press the up and down arrow.
In addition, I can browse other parts of the screen that are not part of the editor and I cannot insert text in the file.
To correct the problem I need to press the alt and escape keys or use the orca command "switch between browse mode and focus mode".
@isidorn When time permits, would it be possible to check?

VS Code version: Code - Insiders 1.44.0-insider (0926b534553f3ff2f0cff0a741550a97df199706, 2020-03-23T08:04:33.593Z)
OS version: Linux x64 5.5.10-arch1-1

**Comment by isidorn** [comment]
@jvesouza I tried your steps, and very time I do the "go to definition" Orca definetly reads too much (like the full file name and the vscode title) however when I do up/down navigation all works just fine for me.
In step 6, when you say press application key you mean `shift + F10` to trigger context menu? Becuase any way I trigger the go to definition I can not reproduce your issue.

fyi @ajborka @joanmarie

**Comment by joanmarie** [comment]
Taking a look.

**Comment by jvesouza** [comment]
> In step 6, when you say press application key you mean `shift + F10` to trigger context menu? Becuase any way I trigger the go to definition I can not reproduce your issue.
Yes, > shift+f10.

**Comment by joanmarie** [comment]
I can confirm this. Something is causing Orca to enter browse mode after returning to the editor from the menu. Probably an Orca bug.

**Comment by ajborka** [comment]
@isidorn  This is partially confirmed. The current filename is read out by Orca after pressing enter on go to definition. However, it might be worth reading the line of text the cursor is placed on. This way, we get context of where we end up in the file. @joanmarie  I can't confirm if Orca leaves focus mode or not. To me, the experience is as though focus mode is always turned on. Using Orca master, Ubuntu 20.04, vscode 1.44 insiders.

**Comment by jvesouza** [comment]
@ajborka What happens if after pressing the enter key you press the orca_key + a? 
Assuming this combination is associated with the switch between browse mode and focus mode command.

**Comment by ajborka** [comment]
@jvesouza  the entire report is confirmed.

**Comment by joanmarie** [comment]
@isidorn when the menu is destroyed and focus returns to the editor, I'm not seeing any state-changed:focused events for the editor or its descendants; just a window:activate event for VSCode*. However, when I use Chromium and a textarea or a contenteditable div, I not only get the state-changed:focused event for the the window, I get one from the document or a descendant therein.

It's the state-changed:focused event which causes Orca to:
1. Read the line
2. Return to focus mode

The only event I am seeing from VSCode in this scenario after the window:activate event is the object:text-caret-moved event from the entry. While I could probably use that to address this issue, I'm curious as to why the normal focus-related event is absent. Any ideas?

*Note: At the present time, when Orca returns to the context menu from any application, Orca receives a window:activate event which causes it to present things like the window title. Perhaps that is too chatty of Orca, but it has nothing to do with VSCode/Chromium specifically. So let's set that issue aside in the context of this bug.

**Comment by jvesouza** [comment]
@joanmarie  @isidorn The curious thing for me is that if I execute the same command by pressing f12 instead of executing it from the context menu, the problem does not seem to happen.

**Comment by ajborka** [comment]
@isidorn , @jvesouza , @joanmarie  . This makes sense. From what I understand, the context menu forces Orca to browse mode. Since there isn't a way for Orca to know if it needs focus mode, it stays in browse mode. In the instance of just pressing f12, Orca has no need to leave focus mode because nothing in the source code has told it to do something else.

**Comment by isidorn** [comment]
Thanks all for providing details.

This smells like an Electron native context menu issue. So the gist of the whole discussion is the following:
VS Code does not send a `focused` event on native context menu close .

I am not sure when Chrome decides to send the focused event. But I do remember that native context menus used to block the renderer process. I do not think that is the case still since I see the cursor blinking when context menu is invoked. @deepak1556 would know better thus assigning also to him.

**Workaround**: set 	`"window.titleBarStyle": "custom"` to use our custom VS Code menus. Then this issue does not reproduce.

**Comment by jvesouza** [comment]
I can  confirm that if I set "window.titleBarStyle": "custom" in the settings.json file, the problem can not be reproduced.

**Comment by joanmarie** [comment]
> I am not sure when Chrome decides to send the focused event.

https://cs.chromium.org/chromium/src/ui/accessibility/platform/ax_platform_node_auralinux.cc?type=cs&q=OnWindowActivated&l=3390

**Comment by ajborka** [comment]
Confirmed fixed when using custom titlebar styles.

**Comment by isidorn** [comment]
@joanmarie thanks for the code pointer. @deepak1556 notified me he will look into this in debt week (start of april). Thus assigning to April so we do not loose track of this.

**Comment by isidorn** [comment]
Leaving this up to @deepak1556 
Deepak let me know if you need help here from my side. Thanks!

**Comment by isidorn** [comment]
@deepak1556 friendly ping for this. Are we already including a fix for this maybe via our electron updates?

**Comment by jvesouza** [comment]
I tested using build exploration and the problem seems to persist.

Version: 1.51.0-exploration
Commit: 1dfc92ea9b990fad440fcf12ce46cb85897861ce
Date: 2020-10-22T06:37:58.796Z
Electron: 11.0.0-beta.12
Chrome: 87.0.4280.11
Node.js: 12.18.3
V8: 8.7.220-electron.0
OS: Linux x64 5.9.6-arch1-1

**Comment by isidorn** [comment]
@jvesouza thanks!

**Comment by jvesouza** [comment]
I revisited this bug and noticed that there was a change in its behavior. When I press shift+f10 with the cursor over a variable name and choose go to definition, orca announces the filename and announces that a variable has been found. After that I can browse the file without any problems, unlike before.

In my configuration window.titleBarStyle is not set to custom.

In my opinion this can be closed.

**Comment by isidorn** [comment]
@jvesouza music to my ears, then let's close this. Thanks for re-testing this issue.
